internal/exec/stack_processor_utils.go

```diff
@@ -1741,11 +1740,48 @@ func FindComponentDependenciesLegacy(
 	return unique, nil
 }
 
+// resolveRelativePath checks if a path is relative to the current directory and if so,
+// resolves it relative to the current file's directory. It ensures the resolved path
+// exists within the base path.
+func resolveRelativePath(path string, currentFilePath string) string {
+	if path == "" {
+		return path
+	}
+
+	// Store original path format (using \ or /)
+	originalPathSeparator := "/"
+	if strings.Contains(path, "\\") {
+		originalPathSeparator = "\\"
+	}
```

Comment:
Something about this has code smell. If we're properly using the native filepath methods, this shouldn't be necessary.